controllers/metal3.io/host_state_machine_test.go

```diff
@@ -289,6 +289,47 @@ func TestErrorCountCleared(t *testing.T) {
 	}
 }
 
+func TestErrorClean(t *testing.T) {
+
+	tests := []struct {
+		Scenario   string
+		Host       *metal3v1alpha1.BareMetalHost
+		SecretName string
+	}{
+		{
+			Scenario: "clean-after-registration-error",
+			Host: host(metal3v1alpha1.StateInspecting).
+				setStatusError(metal3v1alpha1.OperationalStatusError, metal3v1alpha1.RegistrationError, "some error", 1).
+				build(),
+		},
+		{
+			Scenario: "clean-after-creds-change",
+			Host: host(metal3v1alpha1.StateReady).
+				setStatusError(metal3v1alpha1.OperationalStatusError, metal3v1alpha1.InspectionError, "some error", 1).
+				build(),
+			SecretName: "NewCreds",
+		},
+	}
+	for _, tt := range tests {
+		t.Run(tt.Scenario, func(t *testing.T) {
+			prov := &mockProvisioner{}
+			hsm := newHostStateMachine(tt.Host, &BareMetalHostReconciler{}, prov, true)
+
+			info := makeDefaultReconcileInfo(tt.Host)
+			if tt.SecretName != "" {
+				info.bmcCredsSecret.Name = tt.SecretName
+			}
+
+			hsm.ReconcileState(info)
+
+			assert.Equal(t, tt.Host.Status.OperationalStatus, v1alpha1.OperationalStatusOK)
+			assert.Empty(t, tt.Host.Status.ErrorType)
+			assert.Empty(t, tt.Host.Status.ErrorMessage)
+			assert.Empty(t, tt.Host.Status.ErrorCount)
```

Comment:
`clearError()` no longer resets the ErrorCount since #749 merged. That's why this test was passing earlier (in Andrea's PR) but failing now.
I think we can just get rid of this assertion.